### PR TITLE
add go example

### PR DESCRIPTION
**Pull request by zeitlinger**

### CODE REVIEW

File: examples/java/run.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+set -euo pipefail
```
Comment (zeitlinger) [inline examples/java/run.sh:3]: file diff doesn't show, this was the old version:

```bash
#!/bin/bash

cd example-app
if [[ ! -f ./target/example-app.jar ]] ; then
    ./mvnw clean package
fi
if [[ ! -f ./opentelemetry-javaagent.jar ]] ; then
    curl -sOL https://github.com/open-telemetry/opentelemetry-java-instrumentation/releases/download/v2.0.0/opentelemetry-javaagent.jar
fi
export OTEL_EXPORTER_OTLP_PROTOCOL=grpc
export OTEL_SEMCONV_STABILITY_OPT_IN=http
export OTEL_RESOURCE_ATTRIBUTES="service.name=example-app,service.instance.id=localhost:8080"
java -Dotel.metric.export.interval=500 -Dotel.bsp.schedule.delay=500 -javaagent:opentelemetry-javaagent.jar -jar ./target/example-app.jar
```

Comment (zeitlinger) [reply examples/java/run.sh:3]: in the new version

- removed `export OTEL_SEMCONV_STABILITY_OPT_IN=http` that is no longer needed
- don't reuse old downloaded jar if the version doesn't match

**Review by fstab — APPROVED** [review]
LGTM too, thanks for adding a Go example!

Optional remark: We could add a comment in the OATS-related files with a link to https://github.com/grafana/oats in case people are wondering what these files are for.

**Comment by zeitlinger** [comment]
> I just have one comment/question, do you think it's OK to change the default port for the Go application so it doesn't conflict with what Java uses? That way we can run both at the same time...

@grcevski  the [traffic script](https://github.com/grafana/docker-otel-lgtm/blob/065a49d35afa7cc21cd92d6256d040a678193458/generate-traffic.sh#L3) goes to 8080 - should this script hit all applications?

**Comment by zeitlinger** [comment]
> Hm yeah, I am not sure what's best then. Maybe leave it as is, if it goes to both, then what if a user only boots Java, the script will hit something else or non existent port.
> 
> At least with duplicate port they will see it and adjust if they need to. LGTM with out my suggestion.

I hit all ports - but ignore errors

**Comment by grcevski** [comment]
> > Hm yeah, I am not sure what's best then. Maybe leave it as is, if it goes to both, then what if a user only boots Java, the script will hit something else or non existent port.
> > At least with duplicate port they will see it and adjust if they need to. LGTM with out my suggestion.
> 
> I hit all ports - but ignore errors

Great!

**Pull request merged**